connectivity/drivers/cellular/QUECTEL/BG96/QUECTEL_BG96_CellularStack.cpp

```diff
@@ -157,6 +158,40 @@ nsapi_error_t QUECTEL_BG96_CellularStack::socket_connect(nsapi_socket_t handle,
     return err;
 }
 
+nsapi_error_t QUECTEL_BG96_CellularStack::get_ip_address(SocketAddress *address)
+{
+    if (!address) {
+        return NSAPI_ERROR_PARAMETER;
+    }
+    _at.lock();
+
+    _at.cmd_start_stop("+QIACT", "?");
+    _at.resp_start("+QIACT:");
+
+    int32_t context_type = 0;
+
+    if (_at.info_resp()) {
+        _at.skip_param(); // ID
+        _at.skip_param(); // State
+
+        context_type = _at.read_int();
+        if (context_type == 1) {
+            _stack_type = IPV4_STACK;
+        } else if (context_type == 2) {
+            _stack_type = IPV6_STACK;
+        }
```

Comment:
either it's physically impossible for other values to be here or there should be an else clause

Reply:
Only allowed values are 1 and 2 according to the BG95 TCP/IP application note.

Reply:
in which case the second `else if` is redundant and could just be an `else`, would also make my comment below moot as well but it's all nitpicking really